woocommerce-payments.php

```diff
@@ -355,16 +356,7 @@ function wcpay_get_jetpack_idc_custom_content(): array {
 	$urls = Automattic\Jetpack\Identity_Crisis::get_mismatched_urls();
 	if ( false !== $urls ) {
 		$current_url = untrailingslashit( $urls['current_url'] );
-		/**
-		 * Undo the reverse the Jetpack IDC library is doing since we want to display the URL.
-		 *
-		 * @see https://github.com/Automattic/jetpack-identity-crisis/blob/trunk/src/class-identity-crisis.php#L471
-		 */
-		$idc_sync_error = Automattic\Jetpack\Identity_Crisis::check_identity_crisis();
-		if ( is_array( $idc_sync_error ) && ! empty( $idc_sync_error['reversed_url'] ) ) {
-			$urls['wpcom_url'] = strrev( $urls['wpcom_url'] );
-		}
-		$wpcom_url = untrailingslashit( $urls['wpcom_url'] );
```

Comment:
This actually reverts PR #8033 as we no longer need this anymore. 

Looking at [Jetpack code](https://github.com/Automattic/jetpack/blob/06cda771335eca520400af8dd2c13a3769b75e9f/projects/packages/connection/src/identity-crisis/class-identity-crisis.php#L501-L506), we expect that Jetpack should return reversed URLs but it seems Jetpack is no longer do so.

Reply:
The reversal code was the same since it was introduced so what users reported for WCPay and reverted URLs shouldn't be possible.
Could it be that there was a conflict of IDC versions? For example, users installed a newer Jetpack that had the reversing logic that wrote the IDC domains in reverse, then they disabled it, and what was left was WCPay with an older version of the IDC class with no reversal, so they experienced the strange reversed URL?

Maybe the fix on WCPay side should include the package version check instead of removing it completely?

Reply:
> Could it be that there was a conflict of IDC versions? For example, users installed a newer Jetpack that had the reversing logic that wrote the IDC domains in reverse, then they disabled it, and what was left was WCPay with an older version of the IDC class with no reversal, so they experienced the strange reversed URL?

That could be but I am not entirely sure. 

@oaratovskyi @vladolaru @ismaeldcom - could you have a look at this specific part? For now, removing your fix on #8033 helped to fix the issue; otherwise, now it becomes this issue #8025

Reply:
🤔  That's strange, IIRC the fix should only reverse an already reversed URL.

Reply:
I was able to confirm my theory. You can access a test JN site using this data: 3732b-pb/

I loaded an old WCPay version (pre custom WCPay fix) with a new Jetpack. Since Jetpack created the Safe Mode reversed entry, the old WCPay wasn't able to work with that and displayed a reversed URL. Try dis/re-enabling WCPay on the JN test site for results.
So your fix should only run on WCPay when using identity-crisis package pre-v0.10.6 (see here: https://github.com/Automattic/jetpack-identity-crisis/compare/v0.10.5...v0.10.6). If the connection package is used for idc (since we moved the idc package to connection one), you should be fine.

Reply:
@bindlegirl and @ismaeldcom - thanks for providing more info and the test! If so, the right thing here is just to delete the patch created by PR #8033, i.e. what I am doing it now.